hudi-common/src/test/java/org/apache/hudi/common/table/read/TestHoodieFileGroupReaderBase.java

```diff
@@ -154,10 +154,17 @@ public void testReadFileGroupInMergeOnReadTable(RecordMergeMode recordMergeMode,
     }
   }
 
+  private static Stream<Arguments> logFileOnlyCases() {
+    return Stream.of(
+        arguments(RecordMergeMode.COMMIT_TIME_ORDERING, "avro"),
+        arguments(RecordMergeMode.EVENT_TIME_ORDERING, "parquet"),
+        arguments(RecordMergeMode.CUSTOM, "avro"));
+  }
+
   @ParameterizedTest
-  @MethodSource("testArguments")
-  public void testReadLogFilesOnlyInMergeOnReadTable(RecordMergeMode recordMergeMode, String logDataBlockFormat, boolean populateMetaFields) throws Exception {
-    Map<String, String> writeConfigs = new HashMap<>(getCommonConfigs(recordMergeMode, populateMetaFields));
+  @MethodSource("logFileOnlyCases")
+  public void testReadLogFilesOnlyInMergeOnReadTable(RecordMergeMode recordMergeMode, String logDataBlockFormat) throws Exception {
```

Comment:
If the meta fields are not populated, the in-memory index does not work so I had to add a different set of cases here.

Reply:
@nsivabalan @danny0405 is virtual key not supported with in-memory index?

Reply:
The `updateLocation` method was not called when using virtual keys so the map of current locations was not updated and the records got written to new files

Reply:
I see.  @danny0405 @nsivabalan let's make sure this does not affect MDT since MDT uses virtual keys.